docs/src/literate/incompressible_hyperelasticity.jl

```diff
@@ -0,0 +1,362 @@
+#   \widehat{\Psi}(p) = \sup_{J} p(J - 1) - \frac{\mu}{2}\left(I_1 - 3 \right) + \mu \log(J) - \frac{\lambda}{2}\left( J - 1\right){}^2
+# ```
+# where $J^\star$ is given by
+# ```math
+# J^\star = \arg\max \left[ p(J - 1) - \frac{\mu}{2}\left(I_1 - 3 \right) + \mu \log(J) - \frac{\lambda}{2}\left( J - 1\right){}^2\right]
+# ```
+# which yields
+# ```math
+#   J^\star(p) = \frac{\lambda + p + \sqrt{(\lambda + p)^2 + 4 * \lambda * \mu }}{(2 * \lambda}
+# ```
+# Taking the partial legendre transform of $\widehat{\Psi}$ gives us back the original problem now reformulated as
+# ```math
+#   \min_{u\in\mathcal{K}}\max_{p} \int_\Omega\Psi^{\star} (u, p)
+# ```
+# where $\Psi^\star (u, p)$ is given by
+# ```math
+#   \int_\Omega p (J - J^\star) d\Omega + \int_\Omega \frac{\mu}{2} \left( I_1 - 3\right) d\Omega - \int_\Omega \mu\log(J^\star)d\Omega + \int_\Omega \frac{\lambda}{2}\left( J^\star - 1 \right){}^2
+# ```
+# The euler-lagrange equations are then given by
+# ```math
+#   \int_\Omega \frac{\partial\Psi^\star}{\partial F}:\delta F d\Omega = 0 
+# ```
+# and
+# ```math
+#   \int_\Omega \frac{\partial \Psi^\star}{\partial p}\delta p d\Omega
+# ```
+# where δF = δ∇u = ∇(δu) and δu and δp denote test functions for the displacement and pressure fields respectively.
+# In order to apply Newton's method to the above problem, we need to calculate the respective hessians (tangent),
+# namely, ∂²Ψ*/∂F∂F, ∂²Ψ*/∂p² and ∂²Ψ*/∂F∂p which can be determined conveniently from automatic differentiation. 
+# Hence we only need to define the potential. The rest of the program follows easily. First, we import the respective packages
+
+using Ferrite, Tensors, TimerOutputs, ProgressMeter
+using BlockArrays, SparseArrays, LinearAlgebra
+
+# and the corresponding `struct` to store our material properties.
+
+struct NeoHooke
+    μ::Float64
+    λ::Float64
+end
+
+# We then create a function to generate a simple test mesh on which to compute FE solution
+function importTestGrid()
+    grid = generate_grid(Tetrahedron, (5, 5, 5), zero(Vec{3}), ones(Vec{3}))
+    addfaceset!(grid, "myBottom", x -> norm(x[2]) ≈ 0.0);
+    addfaceset!(grid, "myBack", x -> norm(x[3]) ≈ 0.0);
+    addfaceset!(grid, "myRight", x -> norm(x[1]) ≈ 1.0);
+    addfaceset!(grid, "myLeft", x -> norm(x[1]) ≈ 0.0);
+    return grid
+end
+
+# The function to create corresponding cellvalues for the displacement field `u` and pressure `p`
+# follows in a similar fashion from the `incompressible_elasticity` example
+function create_values(interpolation_u, interpolation_p)
+    ## quadrature rules
+    qr      = QuadratureRule{3,RefTetrahedron}(4)
+    face_qr = QuadratureRule{2,RefTetrahedron}(4)
+
+    # geometric interpolation
+    interpolation_geom = Lagrange{3,RefTetrahedron,1}()
+
+    # cell and facevalues for u
+    cellvalues_u = CellVectorValues(qr, interpolation_u, interpolation_geom)
+    facevalues_u = FaceVectorValues(face_qr, interpolation_u, interpolation_geom)
+
+    # cellvalues for p
```

Comment:
> This is really "just" quasi incompressible and not perfectly incompressible, or am I missing something? If so, then, in order to to avoid confusion, I would suggest to name everything `quasi_incompressible...` or similar.

Yep, exactly. Thanks!!  Nearly- or "quasi"-incompressible hyperelasticity. The formulation should be well behaved for perfectly incompressible too, just that it would need some fixes in the weak form (or rather potential in this case).

Reply:
Just pushed some changes. I have a naive question, do you use a formatter to automatically fix these? I mean single `#` and so on? This is my first time opening a PR with a julia package. Thanks :) !

Reply:
No problem. :) I personally fix these by hand, as I am not aware of tooling here. Ping @koehlerson - maybe you have some more useful tips here?

Reply:
Sounds good.Thanks!